docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,177 @@
+* When starting the server
+
+The first stage involves running the `build` command and set any **build option** available from this command to configure the server.
+
+By looking at the individual configuration options at <@links.server id="all-config"/>, you notice
+that some options are marked with an icon to indicate they are a build option. In other words, they can only be set and only take effect when running the `build` command.
+You can also check all the options that require a build by looking at the `help` message of the `build` command:
+
+<@kc.build parameters="--help"/>
+
+The `build` command is responsible for producing an immutable and optimized server image, which is similar to building a container image. In addition to persisting
+any build option you have set, this command also performs a series of optimizations to deliver the best runtime when starting and running the server. As a result,
+a lot of processing that would usually happen when starting and running the server is no longer necessary and the server can start and run faster.
+
+Some optimizations performed by the `build` command are:
+
+* Closed-world assumption about installed providers, which means that no need to re-create the registry every time the server starts
+* Configuration files are pre-parsed to reduce IO when starting the server
+* Database specific resources are configured and prepared to run against a specific database vendor
+* By persisting build options into the server image, the server does not perform any additional step to interpret configuration options and (re)configure itself
+
+Once you run the `build` command, you won't need to set the same build options again when starting the server.
+
+.First stage, run the `build` command to set any build option
+<@kc.build parameters="--db=postgres"/>
+
+The second stage involves starting the server using any **configuration option** available from the `start` command.
+
+.Second stage, run the `start` command to set any configuration option when starting the server
+<@kc.start parameters="--db-url-host=keycloak-postgres --db-username=keycloak --db-password=change_me --hostname mykeycloak.acme.com"/>
+
+At this stage, you are free to set any value you want to any of the configuration options.
+
+== Configuring the server for an optimal startup time
+
+In addition to the optimizations performed when you run the `build` command, you might want to avoid using CLI options when running the
+`start` command in favor of using environment variables or configuration options from the `conf/keycloak.conf` file.
+
+.Set any build option
+<@kc.build parameters="--db=postgres"/>
+
+.Set any configuration option to `conf/keycloak.conf`
+```
+db-url-host=keycloak-postgres
+db-username=keycloak
+db-password=change_me
+hostname mykeycloak.acme.com
+```
+
+.Start the server
+<@kc.start/>
+
+By following that approach, the server is going to skip some steps at startup and start faster.
```

Comment:
This section doesn't make any sense to me as it is suggesting properties in `conf/keycloak.conf` are somehow optimised compared to environment variables and configuration options, which is not the case. You've already previously mentioned that people can mix/match what config source they want, as well as the build stuff, so I would just remove this section completely as it's just duplication.

Reply:
@stianst It is the case. By not using CLI options the server skips some additional steps related to CLI init/parsing.

I'm only using the configuration file as an example, but the section also mentions using env vars.

I don't think we should remove this section.

Reply:
Got a bit confused when reading the source, when I read it in html file format it made more sense